### PR TITLE
[testutil] fixes #913 Hexdumps for P2P messages

### LINKED ISSUE #913: [examples] Automatically generate hex dump of gnet messages
Script input should be a set of message objects , which are meant to be serialized (binary) and printed as an hex dump , similar to the snippet below . Beware of fields excluded from serialization (e.g. message context) .

```
02000000 ........................... Block version: 2

b6ff0b1b1680a2862a30ca44d34
6d9e8
910d334beb48ca0c0000000000000000 ... Hash of previous block's header
9d10aa52ee949386ca9385695f04ede2
70dda20810decd12bc9b048aaab31471 ... Merkle root

24d95a54 ........................... Unix time: 1415239972
30c31b18 ........................... Target: 0x1bc330 * 256**(0x18-3)
fe9f0864 ........................... Nonce
```

### PR DESCRIPTION
**Pull request by olemis**
refs #913 

Changes:
- Hex dump for messges registered against daemont/gnet
- Test examples

### CODE REVIEW

**Comment by olemis** [comment]
@stdevEdu example output is not finished . I only added one as example . Please finish them all avoiding as much as possible generation of test data so that output be deterministic .

**Comment by stdevEdu** [comment]
Removed Examples for Ping & Pong messages, as these are not to be serialized

File: src/daemon/message_hexdump.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
+package daemon
+
+import (
+	"fmt"
+	"strconv"
+
+	"strings"
+)
+
+//var registered = false
```
Comment (gz-c) [inline src/daemon/message_hexdump.go]: delete

File: src/daemon/message_hexdump.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+/*
+func GenerateRandomSha256() cipher.SHA256 {
```
Comment (gz-c) [inline src/daemon/message_hexdump.go]: delete

File: src/daemon/message_hexdump.go [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+	for i := 0; i < len(buffer); i++ {
+		if len(hexBuff[i]) == 1 {
+			hexBuff[i] = "0" + hexBuff[i]
+		}
+	}
+	fmt.Println(getSliceContentsString(hexBuff, offset), name)
+}
+
+/*
+func HexDump(message gnet.Message) string {
```
Comment (gz-c) [inline src/daemon/message_hexdump.go]: delete if unused

File: src/daemon/messages_test.go [hunk 4]
```diff
@@ -1 +1,516 @@
...
+	// 0x000c | 01 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
+	// 0x001c | 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
+	// 0x002c | 00 00 00 00 ....................................... Txns#0
+	// 0x0034 | 01 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
+	// 0x0044 | 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
+	// 0x0054 | 00 00 00 00 ....................................... Txns#1
+	// 0x004c |
+}
+
+func TestSucceed(t *testing.T) {
```
Comment (gz-c) [inline src/daemon/messages_test.go]: delete

**Comment by olemis** [comment]
Superseded by #1241

**Pull request closed without merging**